### DISCUSSION TITLE
Attaching an item with attachments

### DISCUSSION THREAD

**Discussion by dmsfabiano** [ecederstrand/exchangelib#940] [Q&A] [answered]
I may be missing something from the documentation; but say I have a given item (which may have attachments inside of it):

`item = email_account.inbox.filter(message_id__in=[message_id]).only('subject','id','message_id',author','sender',references','attachments')[0]`

say now I want to attach this given item, as an attachment to a msg that i had prepared before
`msg.attach([ItemAttachment(item=item)]`
`msg.send_and_save()`

This yields to:
`exchangelib.errors.ErrorRequiredPropertyMissing: Required property is missing. (field: <t:ExceptionFieldURI xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types" xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" FieldURI="attachment:Name"/>)`

I am assuming I am using this wrong, I couldn't find the proper usage in the documentation. Thank you in advance!

**Comment by ecederstrand** [comment, marked as answer]
The `name` field is empty on one of the attachments you are trying to save. You need to give your attachment a name.
